app/views/admin/dashboard.html.erb

```diff
@@ -88,14 +92,17 @@
         <div class="card-body users-list p-0">
           <!-- /.table-responsive -->
           <ul class="users-list clearfix">
-
-          <% @recent_users.each do |user| %>
-            <li>
-              <%= image_tag gravatar_url(user.email, 32) %>
-              <%= link_to user.name, admin_user_path(user), class: "users-list-name" %>
-              <span class="users-list-date"><%= time_ago_in_words(user.created_at) %></span>
-            </li>
-          <% end %>
+            <% @recent_users.each do |user| %>
+              <li>
+                <%= image_tag gravatar_url(user.email, 32) %>
+                <% if !user.name.nil? %>
+                  <%= link_to user.email, edit_admin_user_path(user), class: "users-list-name" %>
+                <% else %>
+                  <%= link_to user.name, edit_admin_user_path(user), class: "users-list-name" %>
+                <% end %>
```

Comment:
I decided to create a helper in order to do it, since we have a default value for the name field. So, when the name it is equal to "Name Not Provided" we display email otherwise the real name